color_bar_tool.coffee

```diff
@@ -0,0 +1,164 @@
+_ = require "underscore"
+
+SelectTool = require "models/tools/gestures/select_tool"
+BoxAnnotation = require "models/annotations/box_annotation"
+ColorBar = require "models/annotations/color_bar"
+LinearColorMapper = require "models/mappers/linear_color_mapper"
+p = require "core/properties"
+
+class ColorBarToolView extends SelectTool.View
+
+  initialize: (options) ->
+    super(options)
+    @listenTo(@model, 'change:active', () ->
+      @_toggle_active()
+      @plot_view.request_render()
+    )
+
+  _toggle_active: () ->
+    @model.color_bar.visible = @model.active
+    @model.overlay.update({left: null, right: null, top: null, bottom: null})
+
+  _get_image_dims: () ->
+    renderers = _.values(@plot_view.renderer_views)
+    color_bar_views = _.filter(renderers, (r) -> return r instanceof ColorBar.View)
+    color_bar_view = color_bar_views[0]
+
+    panel_offset = color_bar_view._get_panel_offset() #need to flip y value
+    frame_offset = color_bar_view._get_frame_offset()
+    image_offset = color_bar_view._get_image_offset()
+
+    x_offset = (panel_offset.x + frame_offset.x + image_offset.x)
+    y_offset = -(panel_offset.y + frame_offset.y + image_offset.y)
+
+    image_dims = @model.color_bar._computed_image_dimensions() #height and width attrs
+
+    dims = {
+      x0: x_offset
+      x1: x_offset + image_dims.width
+      y0: y_offset - image_dims.height
+      y1: y_offset
+    }
+
+    return dims
+
+  _validate_point: (e) ->
+    dims = @_get_image_dims()
+
+    if dims.x0 <= e.bokeh.sx <= dims.x1
+      x_hit = true
+    else
+      x_hit = false
+
+    if dims.y0 <= e.bokeh.sy <= dims.y1
+      y_hit = true
+    else
+      y_hit = false
```

Comment:
That's pretty verbose, `y_hit = dims.y0 <= ...` would suffice. The same for `x_hit`.